test/src/main/java/org/apache/accumulo/test/upgrade/UpgradeIT.java

```diff
@@ -0,0 +1,152 @@
+  }
+
+  @Test
+  public void testServersWontStart() throws Exception {
+    // Constants.ZPREPARE_FOR_UPGRADE is created by 'accumulo upgrade --prepare'
+    // which is run when the user wants to shutdown an instance in preparation
+    // to upgrade it. When this node exists, no servers should start. There is
+    // no ability to create this node in ZooKeeper before MAC starts for this
+    // test, so we will create the node after MAC starts, then try to restart
+    // MAC.
+
+    final ZooSession zs = getServerContext().getZooSession();
+    final ZooReaderWriter zrw = zs.asReaderWriter();
+    final String upgradePath = Constants.ZPREPARE_FOR_UPGRADE;
+    zrw.putPersistentData(upgradePath, new byte[0], NodeExistsPolicy.SKIP);
+
+    getCluster().stop();
+
+    getCluster().getClusterControl().startAllServers(ServerType.ZOOKEEPER);
+
+    final ZooReader zr = zs.asReader();
+    Wait.waitFor(() -> zr.getChildren(Constants.ZCOMPACTORS).isEmpty());
+    Wait.waitFor(() -> zr.getChildren(Constants.ZGC_LOCK).isEmpty());
+    Wait.waitFor(() -> zr.getChildren(Constants.ZMANAGER_LOCK).isEmpty());
+    Wait.waitFor(() -> zr.getChildren(Constants.ZSSERVERS).isEmpty());
+    Wait.waitFor(() -> zr.getChildren(Constants.ZTSERVERS).isEmpty());
+
+    assertThrows(IllegalStateException.class,
+        () -> assertTimeoutPreemptively(Duration.ofMinutes(2), () -> getCluster().start()));
+
+    assertTrue(zr.getChildren(Constants.ZCOMPACTORS).isEmpty());
+    assertTrue(zr.getChildren(Constants.ZGC_LOCK).isEmpty());
+    assertTrue(zr.getChildren(Constants.ZMANAGER_LOCK).isEmpty());
+    assertTrue(zr.getChildren(Constants.ZSSERVERS).isEmpty());
+    assertTrue(zr.getChildren(Constants.ZTSERVERS).isEmpty());
+
+    // Validate the exception from the servers
+    List<String> args = new ArrayList<>();
+    args.add("--props");
+    args.add(getCluster().getAccumuloPropertiesPath());
+    IllegalStateException ise = assertThrows(IllegalStateException.class,
+        () -> new ServerThatWontStart(args.toArray(new String[0])));
+    assertTrue(ise.getMessage()
+        .startsWith("Instance has been prepared for upgrade to a minor or major version"));
+
+  }
+
+  @Test
+  public void testUpgradeManagerFailure() throws Exception {
+    // Test that the Manager fails when an upgrade is needed,
+    // but UpgradeProgressTracker.initialize was never called.
+
+    getCluster().stop();
+    getCluster().getClusterControl().startAllServers(ServerType.ZOOKEEPER);
+
+    ServerContext ctx = getCluster().getServerContext();
+    UpgradeUtilIT.downgradePersistentVersion(ctx);
+
+    List<String> args = new ArrayList<>();
+    args.add("--props");
+    args.add(getCluster().getAccumuloPropertiesPath());
+    try (TestManager mgr = new TestManager(args.toArray(new String[0]))) {
+      IllegalStateException ise = assertThrows(IllegalStateException.class, () -> mgr.runServer());
+      assertTrue(ise.getMessage().startsWith("initialize not called, "));
```

Comment:
Nice test.  Went looked at the code that throws the exception because I did not know what `initialize not called` meant.   Still not sure after looking at the code.  Wondering if its related to an older version of these changes.  If so, should it be something like :

```
      throw new IllegalStateException("Upgrade node started, " + Constants.ZUPGRADE_PROGRESS
          + " node does not exist. Did you run 'accumulo upgrade --start'?");
```

Reply:
It means that the `UpgradeProgressTracker.initiaize` method was not called to seed the node in ZooKeeper. `initialize` is only called by `accumulo upgrade --start`. I'm fine with changing the message.

Reply:
Fixed in 2762f78